### PR TITLE
GEODE-6677: Fix compile errors

### PR DESCRIPTION
**Pull request by jmelchio**
Remove duplicate switch section

Authored-by: Joris Melchior <joris.melchior@gmail.com>

Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [ ] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [ ] Is your initial contribution a single, squashed commit?

- [ ] Does `gradlew build` run cleanly?

- [ ] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, check Concourse for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

File: geode-core/src/distributedTest/java/org/apache/geode/internal/cache/execute/FunctionRetryTestBase.java [hunk 1]
```diff
@@ -262,11 +262,6 @@ private Execution getExecutionTarget(ExecutionTarget executionTarget) {
         execution = FunctionService.onServers(clusterStartupRule.getClientCache().getDefaultPool())
             .setArguments(200);
         break;
-      case SERVERS_REGION_SERVICE:
```
Comment (agingade) [inline geode-core/src/distributedTest/java/org/apache/geode/internal/cache/execute/FunctionRetryTestBase.java]: Why is this case statement is removed?

Comment (jmelchio) [reply geode-core/src/distributedTest/java/org/apache/geode/internal/cache/execute/FunctionRetryTestBase.java]: @agingade It was a duplicate as far as we could tell.

File: geode-core/src/main/java/org/apache/geode/cache/execute/Execution.java [hunk 2]
```diff
@@ -51,7 +51,8 @@
    *         {@link FunctionService#onRegion(org.apache.geode.cache.Region)}
    * @since GemFire 6.0
    */
-  Execution<ArgumentT, ReturnT, AggregatorT> withFilter(Set<?> filter);
+  <ArgumentT, ReturnT, AggregatorT> Execution<ArgumentT, ReturnT, AggregatorT> withFilter(
```
Comment (Bill) [inline geode-core/src/main/java/org/apache/geode/cache/execute/Execution.java]: I don't understand why the type parameters are required at the beginning of this declaration—they are already defined on the class.

Comment (jmelchio) [reply geode-core/src/main/java/org/apache/geode/cache/execute/Execution.java]: @Bill they are required when you try to use them to chain things. We didn't have them originally and that is what broke your code.

File: geode-core/src/distributedTest/java/org/apache/geode/internal/cache/execute/FunctionRetryTestBase.java (hunk 1 above)
Comment (jmelchio) [reply geode-core/src/distributedTest/java/org/apache/geode/internal/cache/execute/FunctionRetryTestBase.java]: @agingade but on second look I see it is `SERVER` vs. `SERVERS`. I'll push a fix.

File: geode-core/src/main/java/org/apache/geode/cache/execute/Execution.java (hunk 2 above)
Comment (Bill) [reply geode-core/src/main/java/org/apache/geode/cache/execute/Execution.java]: I kind of understand the effect. I don't know why it works though.

Do you have a reference that explains why it works? I'd love that for my mental toolkit.

I've seen situations where a complex expression had to be broken in two for the compiler to understand it, like this:

https://e.printstacktrace.blog/java-type-inference-generic-methods-chain-call/
https://github.com/Bill/FJ/blob/master/src/test/java/com/thoughtpropulsion/fj/ClosedTest.java#L36

@agingade found that without the changes in this PR, he was able to eliminate the compiler errors in `FunctionRetryTestBase.java` by changing e.g. this:

```java
      case REGION:
        execution =
            FunctionService.onRegion(clusterStartupRule.getClientCache().getRegion(regionName))
                .setArguments(200);
```

to this:

```java
      case REGION:
        final Execution<Integer, Long, List<Long>>
            tmp =
            FunctionService.onRegion(clusterStartupRule.getClientCache().getRegion(regionName));
        execution = tmp.setArguments(200);
        break;
```

**Comment by dschneider-pivotal** [comment]
is GEODE-6677 correct?

**Comment by dschneider-pivotal** [comment]
Since develop is broken I would recommend reverting the fix that broke it and then resubmit that fix with this work instead of it being in two different tickets.

**Comment by jinmeiliao** [comment]
It's actually the latest checkin that broke it. this fix is actually makes it more resilient for method chaining

File: geode-core/src/main/java/org/apache/geode/cache/execute/Execution.java (hunk 2 above)
Comment (Bill) [reply geode-core/src/main/java/org/apache/geode/cache/execute/Execution.java]: PR looks good to me. I'll educate myself on the details of _why_ this fix works, on my own time after the PR is merged.

File: geode-core/src/distributedTest/java/org/apache/geode/internal/cache/execute/FunctionRetryTestBase.java (hunk 1 above)
Comment (agingade) [reply geode-core/src/distributedTest/java/org/apache/geode/internal/cache/execute/FunctionRetryTestBase.java]: Also it should not be referring to GEODE-6677. If there is a need to backport or analyze the changes for GEODE-6588; this should be part of that.

File: geode-core/src/main/java/org/apache/geode/cache/execute/Execution.java (hunk 2 above)
Comment (jmelchio) [reply geode-core/src/main/java/org/apache/geode/cache/execute/Execution.java]: @Bill We fixed the issues earlier by breaking things up but when looking at it with @jinmeiliao today we found that the approach we take with `static` methods works on `instance` methods as well.

In that case the `Type` parameters you put before the return type tell the compiler to look at the receiver of the return object for type parameters and infer them from there.

Comment (Bill) [reply geode-core/src/main/java/org/apache/geode/cache/execute/Execution.java]: Thanks @jmelchio I'll have to ponder that. Here is some closely related info: https://blog.jooq.org/2013/11/25/a-lesser-known-java-8-feature-generalized-target-type-inference/

Comment (Bill) [reply geode-core/src/main/java/org/apache/geode/cache/execute/Execution.java]: This PR introduces a new bug:

https://issues.apache.org/jira/browse/GEODE-6884

Comment (jmelchio) [reply geode-core/src/main/java/org/apache/geode/cache/execute/Execution.java]: Yes, more experiments have shown that there are holes when try to chain. Need to go back to drawing board.

**Pull request merged**